I've done terrible things

UnleashMe opened the pull request.

UnleashMe committed:
6

UnleashMe committed:
gi

UnleashMe committed:
Merge pull request #1 from UnleashMe/master

Master

falloutfire commented on app/src/main/java/com/example/neverpidor/data/MenuItemsRepository.kt:
```diff
@@ -0,0 +1,140 @@
+import com.example.neverpidor.network.NetworkLayer
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+
+class MenuItemsRepository {
+
+    suspend fun getBeerById(beerId: String): Beer? {
+
+        val request = NetworkLayer.apiClient.getBeerById(beerId)
+
+        if (request.failed) {
+            return null
+        }
+        if (!request.isSuccessful) {
+            return null
+        }
+        return request.body
+    }
+
+    suspend fun getSnackById(snackId: String): Snack? {
+
+        val request = NetworkLayer.apiClient.getSnackById(snackId)
+
+        if (request.failed) {
+            return null
+        }
+        if (!request.isSuccessful) {
+            return null
+        }
+        return request.body
+    }
+    suspend fun getSnacks(): SnackList? {
+
+        val request = NetworkLayer.apiClient.getSnacks()
```
Использовать форматирование

falloutfire commented on app/src/main/java/com/example/neverpidor/ui/MainActivity.kt:
```diff
@@ -0,0 +1,58 @@
+    private lateinit var appBarConfiguration: AppBarConfiguration
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        _binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+/*
+        setSupportActionBar(binding.toolbar)*/
+
+       // supportActionBar?.setDisplayHomeAsUpEnabled(true)
+       /* val toggle = ActionBarDrawerToggle(
+            this,
+            binding.drawerLayout,
+            binding.toolbar,
+            R.string.app_name,
+            R.string.app_name
+        )
+        binding.drawerLayout.addDrawerListener(toggle)*/
+
+        val navHostFragment = supportFragmentManager.findFragmentById(R.id.container) as NavHostFragment
+        navController = navHostFragment.navController
+        appBarConfiguration = AppBarConfiguration(navController.graph, binding.drawerLayout)
+     //   NavigationUI.setupActionBarWithNavController(this, navController)
+        binding.navView.setupWithNavController(navController)
+        setupActionBarWithNavController(navController, appBarConfiguration)
+
+   /*     toggle.syncState()*/
+    }
+
+    override fun onSupportNavigateUp(): Boolean {
+        return navController.navigateUp(appBarConfiguration)||super.onSupportNavigateUp()
+    }
```
Убрать комменты

falloutfire commented on app/src/main/java/com/example/neverpidor/ui/fragments/itemlist/MenuItemListEpoxyController.kt:
```diff
@@ -0,0 +1,152 @@
+            requestModelBuild()
+        }
+
+
+    override fun buildModels() {
+
+        if (isLoading) {
+            LoadingScreenEpoxyModel().id("Loading").addTo(this)
+            return
+        }
+        when (id) {
+            0 -> {
+                beerList.data.groupBy { it.type }.forEach { map ->
+                    ItemTypeEpoxyModel(map.key, onTypeClick = { string ->
+                        isShown = string
+                    }).id(map.key.hashCode()).addTo(this)
+                    DividerEpoxy(R.color.accent).id(Random.nextDouble(100.0)).addTo(this)
+                    map.value.filter { it.type == isShown }.forEach {
+                        MenuItemEpoxyModel(it, id, onEditClick).id(it.UID).addTo(this)
+                        DividerEpoxy(R.color.black).id(Random.nextDouble(100.0)).addTo(this)
+                    }
+                }
+            }
+            1 -> {
+                snacks.data.groupBy { it.type }.forEach { map ->
+                    ItemTypeEpoxyModel(map.key, onTypeClick = { string ->
+                        isShown = string
+                    }).id(map.key.hashCode()).addTo(this)
+                    DividerEpoxy(R.color.accent).id(Random.nextDouble(100.0)).addTo(this)
+                    map.value.filter { it.type == isShown }.forEach {
+                        MenuItemEpoxyModel(it, id, onEditClick).id(it.UID).addTo(this)
+                        DividerEpoxy(R.color.black).id(Random.nextDouble(100.0)).addTo(this)
+                    }
+                }
+            }
+        }
+    }
+
+    class LoadingScreenEpoxyModel :
+        ViewBindingKotlinModel<ModelLoadingDataScreenBinding>(R.layout.model_loading_data_screen) {
```
0 и 1 заменить на осмысленные названия

UnleashMe committed:
init history

UnleashMe committed:
change git ignore

UnleashMe committed:
changed toast and bugged screenUpdate on add and update methods, added pics

UnleashMe committed:
added shared prefs, starting to implement mapping to domain objects

UnleashMe committed:
added snack pictures,
mapped all network items to domain
added singleFragment
added epoxy to singleFragment
moved all epoxyModels to folders
added lottie to addFragment
added saveState for itemListFragment
added titles to all fragments

todo
hilt
refactoring (strings to resources, deleting imports)

UnleashMe committed:
added snack pictures,
mapped all network items to domain
added singleFragment
added epoxy to singleFragment
moved all epoxyModels to folders
added lottie to addFragment
added saveState for itemListFragment
added titles to all fragments
added hilt
did some refactoring

falloutfire commented on app/src/main/java/com/example/neverpidor/network/NetworkLayer.kt:
```diff
@@ -10,6 +10,8 @@ object NetworkLayer {
+
     private val moshi: Moshi = Moshi.Builder().add(KotlinJsonAdapterFactory()).build()
 
     private val retrofit = Retrofit.Builder()
```
На будущую фичу: можно заинтегрировать логику в зависимости от типа сборки - разный урл будет браться 
https://developer.android.com/studio/build/build-variants
https://www.tanelikorri.com/tutorial/android/set-variables-in-build-gradle/

Ну и переехать на KTS чтоб и в билд файлах был Котлин )

falloutfire commented on app/src/main/java/com/example/neverpidor/ui/epoxy/models/LoadingScreenEpoxyModel.kt:
```diff
@@ -0,0 +1,12 @@
+import com.example.neverpidor.R
+import com.example.neverpidor.databinding.ModelLoadingDataScreenBinding
+import com.example.neverpidor.ui.epoxy.ViewBindingKotlinModel
+
+class LoadingScreenEpoxyModel :
+    ViewBindingKotlinModel<ModelLoadingDataScreenBinding>(R.layout.model_loading_data_screen) {
+    override fun ModelLoadingDataScreenBinding.bind() {
+
+    }
+}
```
это шо такое и че оно пустое ?

falloutfire commented on app/src/main/java/com/example/neverpidor/ui/fragments/addbeer/AddBeerFragment.kt:
```diff
@@ -37,21 +41,25 @@ class AddBeerFragment : BaseFragment() {
+        if (item == R.string.snacks) {
+            supportActionBar?.title = "Добавить закусь"
             binding.volumeTextLayout.isGone = true
             binding.alcTextLayout.isGone = true
```
После это мера ебании еще один отдельный МР на то чтоб все строковые шутки загнать в strings

falloutfire commented on app/src/main/java/com/example/neverpidor/ui/fragments/addbeer/AddBeerFragment.kt:
```diff
@@ -94,7 +102,7 @@ class AddBeerFragment : BaseFragment() {
+        if (id == R.string.beer) {
             val alcPercentageStr = binding.alcEt.text.toString()
             if (alcPercentageStr.isEmpty()) {
                 binding.alcTextLayout.error = "Надо бы добавить хмелю"
```
На подумать как вынести всю логику и валиацию во вью модель

falloutfire commented on app/src/main/java/com/example/neverpidor/ui/fragments/itemlist/MenuItemListViewModel.kt:
```diff
@@ -6,36 +6,41 @@ import androidx.lifecycle.ViewModel
+
     fun getSnacks() = viewModelScope.launch {
         _snacks.postValue(repository.getSnacks())
     }
```
не надо ли случаем выставлять viewModelScope.launch(Dispatchers.IO)?
Если сможешь пруфануть - ок )

falloutfire commented on app/src/main/java/com/example/neverpidor/ui/fragments/itemlist/epoxy/MenuItemListEpoxyController.kt:
```diff
@@ -0,0 +1,95 @@
+                        isShown = isShown + string
+                    }).id(map.key.hashCode()).addTo(this)
+                    DividerEpoxy(R.color.accent).id(Random.nextDouble(100.0)).addTo(this)
+                    map.value.filter { isShown.contains(it.type) }.forEach {
+                        MenuItemEpoxyModel(it, onEditClick, onItemClick).id(it.UID).addTo(this)
+                        //      DividerEpoxy(R.color.black).id(Random.nextDouble(100.0)).addTo(this)
+                    }
```
че за коммент ?

falloutfire commented on app/src/main/java/com/example/neverpidor/ui/fragments/itemlist/epoxy/MenuItemListEpoxyController.kt:
```diff
@@ -0,0 +1,95 @@
+
+    fun getShownState(): Set<String> = isShown
+    fun setShownState(set: Set<String>) {
+        isShown = set
+    }
+
+
+}
```
ну вот коло переносов строк это конечно фе

falloutfire commented on app/src/main/java/com/example/neverpidor/ui/fragments/singleItem/FragmentSingleItem.kt:
```diff
@@ -0,0 +1,114 @@
+    }
+
+    private fun updateUi(item: DomainItem) {
+        supportActionBar?.title = item.name
+        binding.progressBar.isGone = true
+        binding.alcoholPercentageText.isVisible = true
+        binding.volumeText.isVisible = true
+        binding.price.isVisible = true
+        binding.description.isVisible = true
+        binding.imageView.isVisible = true
+        binding.titleText.isVisible = true
+        binding.recyclerView.isVisible = true
+        item.image?.let {
```
такие места стоит заменять например на 
`
binding.apply {
    it.progressBar.isGone = true
}
`

UnleashMe committed:
moved validation to viewModel
added more DI

falloutfire commented on app/src/main/java/com/example/neverpidor/ui/fragments/addbeer/AddBeerViewModel.kt:
```diff
@@ -35,30 +38,107 @@ class AddBeerViewModel @Inject constructor(
 
     fun getItem(): Int = appSettings.getCurrentItem()
+
+    fun validateFields(
```
Много переменных, в идеале вынести в какую нибудь сущность/модель
